tiledb/sm/query/readers/sparse_index_reader_base.h

```diff
@@ -296,20 +296,49 @@ class SparseIndexReaderBase : public ReaderBase {
    * it is really required to determine if a query is incomplete from the client
    * side of a cloud request.
    */
-  struct ReadState {
-    /** The tile index inside of each fragments. */
-    std::vector<FragIdx> frag_idx_;
+  class ReadState {
+   public:
+    /* ********************************* */
+    /*     CONSTRUCTORS & DESTRUCTORS    */
+    /* ********************************* */
 
-    /** Is the reader done with the query. */
-    bool done_adding_result_tiles_;
+    ReadState() = delete;
 
-    ReadState() {
+    ReadState(size_t frag_idxs_len)
+        : frag_idx_(frag_idxs_len) {
     }
 
     ReadState(std::vector<FragIdx>&& frag_idx, bool done_adding_result_tiles)
         : frag_idx_(std::move(frag_idx))
         , done_adding_result_tiles_(done_adding_result_tiles) {
     }
+
+    /* ********************************* */
+    /*              GETTERS              */
+    /* ********************************* */
+
+    const bool& done_adding_result_tiles() const {
+      return done_adding_result_tiles_;
+    }
+
+    bool& done_adding_result_tiles() {
+      return done_adding_result_tiles_;
+    }
+
+    std::vector<FragIdx>& frag_idx() {
```

Comment:
Let's see if we can make:
void set_frag_idx(unsigned, uint64_t)
const std::vector<FragIdx>& frag_idx() const